rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ReportsJdbcRepository.scala

```diff
@@ -382,12 +380,54 @@ class ReportsJdbcRepository(jdbcTemplate : JdbcTemplate) extends ReportsReposito
       }
     }
 
+
+    /*
+     * here, we may have several runs with same nodeId/timestamp
+     * In that case, we need to keep the one with a configId, if such
+     * exists.
+     */
+    def distinctRuns(seq: Seq[AgentRun]): Seq[AgentRun] = {
+      //that one is for a list of agentRun with same id
+      def recDisctinct(runs: List[AgentRun]): AgentRun = {
+        runs match {
+          case Nil => throw new IllegalArgumentException("Error in code: distinctRuns methods should never call the recDistinct one with an empty list")
+          //easy, most common case
+          case h :: Nil => h
+          case a :: b :: t =>
+            if(a == b) {
+              recDisctinct(a :: t)
+            } else (a, b) match {
```

Comment:
there is no order in the list, so how can you ensure it would dedup :
run1, run56, run47, run1